Issue/41

jwokaty opened the pull request:
Incorporates quality control into the files. Currently devel includes unreviewed studies. This PR checks validity of data against https://docs.google.com/document/d/1d8MPEJlgu4EeKDvfZSD6jvo1fz7pVTseHzwr-eyrIhQ/edit?tab=t.0. As a consequence, the 'bleeding edge' would no longer contain all studies. This reduces the current number of rows in full_dump.csv from about 8865 to about 3200.

Alternatively, if it is desirable to continue having a bleeding edge with all, including unreviewed studies, we could make a branch for the release version, like Bioconductor. We'd have to update the action to accommodate this.  I would appreciate your feedback.

Commit:
Add quality control script

Commit:
Incorporate checks into dump_release.R

Commit:
Remove adjustments from troubleshooting

Commit:
Make test for NA or blank positive

lwaldron commented:
I think "bleeding edge" should include all studies. Filtering incomplete or unreviewed studies can be done downstream if needed, using the "State" and "Reviewer" fields.

Commit:
Add validate parameter produce validated files, add inputs action

Commit:
Add additional packages

Commit:
Fix quotes

Commit:
Fix additional quotes

Commit:
Fix dependencies installation

Commit:
Notify if running validation

Commit:
Update PR body message

Commit:
Make validate logical

Commit:
Change PR to draft

jwokaty commented:
I removed validation on `BSDB ID` and `countries` on @cmirzayi 's recommendation and added two inputs to the GHA: `branch` and `validate`. If `validate` is true and a `branch` name is provided, it will validate the data, create a new branch, and create a draft PR so that a knowledgeable reviewer can inspect the files. The PR should not be merged because we don't want to overwrite the devel branch. It can just be closed and any undesirable branches can be deleted if the candidate files are rejected. The default behavior of the GHA maintains the "bleeding edge" current behavior of pushing to devel without additional validation restricting the data.
